scripts/init.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+set -e
+
+# Abort if `venv` folder already exists
+if [ -d ".venv" ]; then
+    echo ".venv/ folder already exists. Deactivate your venv and remove .venv/ folder."
+    exit 1
+fi
+
+# Create and activate a new virtual environment
+python3 -m venv .venv
+source .venv/bin/activate
+
+# Upgrade pip and install all projects and their dependencies
+pip install --upgrade pip
+pip install -e '.[all]'
```

Comment:
I don't have a strong opinion here, but I would say that if you have these instructions in your README and CONTRIBUTION files, it's better to not replicate it in this init script. I mean both works, even though I tend to prefer somebody explaining me the steps that I copy and paste from the README, rather than running blindly a script that can do anything unless I open it and review it.

Again, I don't have a strong opinion so both works for me.

Side note: I made [an exception](https://github.com/palazzem/ha-econnect-alarm/blob/main/scripts/download_fixtures.sh) in my repository, as the steps to configure HA testing in a custom integration are crazy, and I found having an init script as the only way to simplify people life.

Reply:
I see your point and removed it. The changes to the contribution guide and the readme should be enough information to get into development

Reply:
I have changed the Readme and the Contribution Guide according to your suggestions. 

Thank you for your feedback. Although I've been in the coding business for 30 years and thought I had a lot of experience, you're still teaching me a lot here. You learn a lot from such an opensource project (if you want to do it right :))

Reply:
Thank you @wmalgadey , I really appreciate your comment! I think the best part of this job is that we'll never stop learning something new (I'm learning every day, and I learned a lot also from your work and this repository!). I strongly believe that's what makes the job creative/exciting even though we'll still have tough days and challenging projects to push!

Anyway, you are doing an impressive job in keeping this project alive, considering how much is used and is critical to our community! I'll try to help the best I can!